File: backend/apps/documents/models/certificates.py
from django.db import models


class Certificates(models.Model):
    """Maps to `certificates` in db_current."""

    id = models.AutoField(primary_key=True)
    certificate_name = models.CharField(max_length=100)
    certificate_text = models.TextField()
    left_header = models.CharField(max_length=100)
    center_header = models.CharField(max_length=100)
    right_header = models.CharField(max_length=100)
    left_footer = models.CharField(max_length=100)
    right_footer = models.CharField(max_length=100)
    center_footer = models.CharField(max_length=100)
    background_image = models.CharField(max_length=100, blank=True, null=True)
    created_for = models.IntegerField()
    status = models.IntegerField()
    header_height = models.IntegerField()
    content_height = models.IntegerField()
    footer_height = models.IntegerField()
    content_width = models.IntegerField()
    enable_student_image = models.IntegerField()
    enable_image_height = models.IntegerField()
    created_at = models.DateTimeField(blank=True, null=True)
    updated_at = models.DateField(blank=True, null=True)

    class Meta:
        managed = False
        db_table = "certificates"

    def __str__(self):
        return f"Certificates {self.pk}"

File: frontend/src/types/certificates.ts
export interface CertificateTemplate {
  id: number;
  certificate_name: string;
  certificate_text: string;
  left_header: string;
  center_header: string;
  right_header: string;
  left_footer: string;
  right_footer: string;
  center_footer: string;
  background_image: string | null;
  created_for: number;
  status: number;
  header_height: number;
  content_height: number;
  footer_height: number;
  content_width: number;
  enable_student_image: number;
  enable_image_height: number;
  created_at: string | null;
  updated_at: string | null;
}

export type CreateCertificateTemplatePayload = {
  certificate_name: string;
  certificate_text: string;
  left_header?: string;
  center_header?: string;
  right_header?: string;
  left_footer?: string;
  right_footer?: string;
  center_footer?: string;
  background_image?: string | null;
  created_for?: number;
  status?: number;
  header_height?: number;
  content_height?: number;
  footer_height?: number;
  content_width?: number;
  enable_student_image?: number;
  enable_image_height?: number;
};

export type UpdateCertificateTemplatePayload = Partial<CreateCertificateTemplatePayload>;

export interface CertificatePreview {
  certificate_id: number;
  certificate_name: string;
  student_id: number;
  student_name: string;
  left_header: string;
  center_header: string;
  right_header: string;
  left_footer: string;
  center_footer: string;
  right_footer: string;
  certificate_text: string;
  background_image: string | null;
  header_height: number;
  content_height: number;
  footer_height: number;
  content_width: number;
  enable_student_image: number;
  enable_image_height: number;
  student_image: string | null;
}
